cmd/config-migrate.go

```diff
@@ -2585,3 +2588,40 @@ func migrateV30ToV31MinioSys(objAPI ObjectLayer) error {
 	logger.Info(configMigrateMSGTemplate, configFile, "30", "31")
 	return nil
 }
+
+func migrateV31ToV32MinioSys(objAPI ObjectLayer) error {
+	configFile := path.Join(minioConfigPrefix, minioConfigFile)
+
+	ok, data, err := checkConfigVersion(objAPI, configFile, "31")
+	if err == errConfigNotFound {
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("Unable to load config file. %v", err)
+	}
+	if !ok {
+		return nil
+	}
+
+	cfg := &serverConfigV32{}
+	if err = json.Unmarshal(data, cfg); err != nil {
+		return err
+	}
+
+	cfg.Version = "32"
+	if len(cfg.Notify.NSQ) == 0 {
```

Comment:
If you separate the structs this type of logic is not needed, NSQ will get unnecessarily added even during V31 migration due to struct definition. If you keep V32 separate then you are pretty sure that NSQ is empty.